compiler/src/dotty/tools/dotc/config/ScalaSettings.scala

```diff
@@ -160,6 +160,7 @@ private sealed trait WarningSettings:
   val Whelp: Setting[Boolean] = BooleanSetting("-W", "Print a synopsis of warning options.")
   val XfatalWarnings: Setting[Boolean] = BooleanSetting("-Werror", "Fail the compilation if there are any warnings.", aliases = List("-Xfatal-warnings"))
   val WvalueDiscard: Setting[Boolean] = BooleanSetting("-Wvalue-discard", "Warn when non-Unit expression results are unused.")
+  val WNonUnitStatement = BooleanSetting("-Wnonunit-statement", "Warn when block statements are non-Unit expressions.")
```

Comment:
Actually the logic of this flag seems much more complex. It would be good to explain that there are some corner cases, when a block returns a non-unit value but it's considered safe so no warning is raised